File: 027. Remove Element/Remove Element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
         for(vector<int> :: iterator it = nums.begin(); it != nums.end(); ){
            if(*it == val){
                nums.erase(it);
            } else {
                it++;
            }
        }
        return nums.size();
    }
};
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        return distance(nums.begin(), remove(nums.begin(), nums.end(), val));
    }
};
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int index = 0;
        for(int i = 0; i < n; ++i){
            if(nums[i] != val){
                nums[index++] = nums[i];
            }
        }
        return index;
    }
};

File: 027. Remove Element/Remove Element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        n = len(nums)
        index = 0
        for num in nums:
            if num != val:
                nums[index] = num
                index += 1
            
        
        return index
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        while val in nums:
            nums.pop(nums.index(val))
           # nums.remove(val)
            
        
        return len(nums)
